Code/Framework/AzFramework/AzFramework/Visibility/VisibleGeometryBus.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/ComponentBus.h>
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Math/Matrix4x4.h>
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/std/containers/vector.h>
+
+namespace AZ
+{
+    class ReflectContext;
+}
+
+namespace AzFramework
+{
+    //! VisibleGeometry describes details about visible geometry surfaces stored as generic indexed triangle lists
+    struct VisibleGeometry
+    {
+        AZ_TYPE_INFO(VisibleGeometry, "{4B011208-B105-4BC1-A4F3-FD5C44785D71}");
+        AZ_CLASS_ALLOCATOR(VisibleGeometry, AZ::SystemAllocator);
+
+        static void Reflect(AZ::ReflectContext* context);
+
+        AZ::Matrix4x4 m_transform;
+        AZStd::vector<float> m_vertices;
+        AZStd::vector<uint32_t> m_indices;
+    };
+
+    using VisibleGeometryContainer = AZStd::vector<VisibleGeometry>;
+
+    //! Interface for components to provide generic geometric data, potentially for occlusion culling and other systems
+    class VisibleGeometryRequests : public AZ::ComponentBus
+    {
+    public:
+        static void Reflect(AZ::ReflectContext* context);
+
+        //! Returns a container of visible geometry
+        virtual void GetVisibleGeometry(VisibleGeometryContainer& geometryContainer) const = 0;
+
+    protected:
+        ~VisibleGeometryRequests() = default;
+    };
```

Comment:
When looking at this bus, I'm wondering if there's a way to modify the API so that it could also work for terrain or other procedural meshes that might not have a finite set of geometry. For example, should this also use an AABB or something to describe the bounds of the geometry that it wants? Or would that be problematic in the simple model case?

Reply:
We could pass in a bounding box that gets ignored if it is invalid. Terrain could also do its own subdivision to populate the container with multiple tiles or pieces, depending on memory constraints with large terrain. That would be the same with any large mesh. I considered using a visitor pattern instead of storing the geometry in the container but it's going to have to be stored somewhere regardless. With meshes and terrain, we also need to consider how to handle LODs.

Reply:
The issue for terrain without a bounding box is that you could theoretically have one that's procedural and infinite in size, so you literally _can't_ return all of its geometry without bounds. And as you pointed out, even a regular mesh that's giant could still have a prohibitively large amount of geometry. That's why I was wondering if designing the API from the start to partition the data in some way would be useful, instead of just assuming that every listener can return a finite and complete set of data.